app/models/miq_action.rb

```diff
@@ -123,7 +123,7 @@ def self.invoke_actions(apply_policies_to, inputs, succeeded, failed)
 
           name = a.action_type == "default" ? a.name.to_sym : a.action_type.to_sym
           results[name] ||= []
-          results[name] << {:policy_id => p.kind_of?(MiqPolicy) ? p.id : nil, :policy_status => :success, :result => a.invoke(apply_policies_to, inputs)}
+          results[name] << {:policy_id => (p.id if p.kind_of?(MiqPolicy)), :policy_status => :success, :result => a.invoke(apply_policies_to, inputs)}
```

Comment:
I find this particular one weird since it's nested inside of other things.

Reply:
yea I like `p.kind_of?(MiqPolicy) && p.id` syntax

this seems like a partial `if` to me.
dropping

Reply:
update: `a && b` is not the same as `a ? b : nil` - reverting to original stuff